Name "linux" breaks gcc builds

### Describe the bug

Using "linux" as a var name or struct field name causes gcc builds to fail. This is apparently because gcc has a predefined macro named "linux".

### Reproduction Steps

```v
// test.v
fn main() {
	linux := 'hello world'
	println(linux)
}
```
```
v -cc gcc test.v
```

### Expected Behavior

I would expect V to handle this gracefully by choosing a different name to use in the generated C code.

### Current Behavior

```
================== C compilation error (from gcc): ==============
cc: /tmp/v_1000/test.01KAYYXPNRVCHV3HV6DVENV5BS.tmp.c: In function ‘main__main’:
cc: /tmp/v_1000/test.01KAYYXPNRVCHV3HV6DVENV5BS.tmp.c:5207:16: error: expected identifier or ‘(’ before numeric constant
cc:  5207 |         string linux = _S("hello world");
cc:       |                ^~~~~
cc: /tmp/v_1000/test.01KAYYXPNRVCHV3HV6DVENV5BS.tmp.c:144:77: error: expected statement before ‘)’ token
cc:   144 | #define _S(s) ((string){.str=(byteptr)("" s), .len=(sizeof(s)-1), .is_lit=1})
cc:       |                                                                             ^
cc: /tmp/v_1000/test.01KAYYXPNRVCHV3HV6DVENV5BS.tmp.c:5207:24: note: in expansion of macro ‘_S’
cc:  5207 |         string linux = _S("hello world");
cc:       |                        ^~
cc: /tmp/v_1000/test.01KAYYXPNRVCHV3HV6DVENV5BS.tmp.c:5208:26: error: incompatible type for argument 1 of ‘builtin__println’
cc:  5208 |         builtin__println(linux);
...
cc:       |                       ~~~~~~~^
(note: the original output was 18 lines long; it was truncated to its first 12 lines + the last line)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.12 85ff886

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.12 ce3a1611a16cbeecbe4b46f407de84e03eb08668.85ff886
|:-------------------|:-------------------
|OS                  |linux, "Arch Linux"
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
|Memory              |1.37GB/23.22GB
|                    |
|V executable        |/home/scott/builds/v/v
|V last modified time|2025-11-26 02:25:02
|                    |
|V home dir          |OK, value: /home/scott/builds/v
|VMODULES            |OK, value: /home/scott/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/scott
|                    |
|Git version         |git version 2.52.0
|V git status        |weekly.2025.13-1240-g85ff8864
|.git/config present |true
|                    |
|cc version          |cc (GCC) 15.2.1 20251112
|gcc version         |gcc (GCC) 15.2.1 20251112
|clang version       |clang version 21.1.6
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.42

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -111,7 +111,7 @@ ifdef LEGACY
 	rm -rf $(TMPLEGACY)
 	$(eval override LDFLAGS+=-L$(realpath $(LEGACYLIBS))/lib -lMacportsLegacySupport)
 endif
-	$(CC) $(CFLAGS) -std=gnu99 -w -o v1$(EXE_EXT) $(VC)/$(VCFILE) -lm -lpthread $(LDFLAGS) || cmd/tools/cc_compilation_failed_non_windows.sh
+	$(CC) $(CFLAGS) -std=c99 -w -o v1$(EXE_EXT) $(VC)/$(VCFILE) -lm -lpthread $(LDFLAGS) || cmd/tools/cc_compilation_failed_non_windows.sh
 ifdef NETBSD
 	paxctl +m v1$(EXE_EXT)
 endif
@@ -229,4 +229,3 @@ etags:
 
 ctags:
 	./v$(EXE_EXT) -print-v-files cmd/v | grep -v :parse_text| ctags -L -
-

--- a/vlib/v/builder/cc.v
+++ b/vlib/v/builder/cc.v
@@ -11,9 +11,7 @@ import v.vcache
 import term
 
 const c_std = 'c99'
-const c_std_gnu = 'gnu99'
 const cpp_std = 'c++17'
-const cpp_std_gnu = 'gnu++17'
 
 const c_verror_message_marker = 'VERROR_MESSAGE '
 
@@ -465,12 +463,7 @@ fn (mut v Builder) setup_ccompiler_options(ccompiler string) {
 		}
 	}
 	if !v.pref.no_std {
-		if v.pref.os == .linux {
-			ccoptions.source_args << '-std=${c_std_gnu}'
-		} else {
-			ccoptions.source_args << '-std=${c_std}'
-		}
-		ccoptions.source_args << '-D_DEFAULT_SOURCE'
+		ccoptions.source_args << ['-std=${c_std}', '-D_DEFAULT_SOURCE']
 	}
 	$if trace_ccoptions ? {
 		println('>>> setup_ccompiler_options ccompiler: ${ccompiler}')
@@ -545,18 +538,10 @@ fn (mut v Builder) thirdparty_object_args(ccoptions CcompilerOptions, middle []s
 	mut all := []string{}
 
 	if !v.pref.no_std {
-		if v.pref.os == .linux {
-			if cpp_file {
-				all << '-std=${cpp_std_gnu}'
-			} else {
-				all << '-std=${c_std_gnu}'
-			}
+		if cpp_file {
+			all << '-std=${cpp_std}'
 		} else {
-			if cpp_file {
-				all << '-std=${cpp_std}'
-			} else {
-				all << '-std=${c_std}'
-			}
+			all << '-std=${c_std}'
 		}
 		all << '-D_DEFAULT_SOURCE'
 	}

--- a/vlib/v/help/build/build-c.txt
+++ b/vlib/v/help/build/build-c.txt
@@ -341,7 +341,7 @@ see also `v help build`.
       compiler, on the command line, without writing an .rsp file first.
 
    -no-std
-      By default, V passes -std=gnu99(linux)/-std=c99 to the C backend, but some compilers do
+      By default, V passes -std=c99 to the C backend, but some compilers do
       not support that, even though they may be able to compile the produced
       code, or have other options that can be tuned to allow it.
       Passing -no-std will remove that flag, and you can then use -cflags ''

--- a/vlib/v/pref/pref.v
+++ b/vlib/v/pref/pref.v
@@ -227,7 +227,7 @@ pub mut:
 	fatal_errors     bool // unconditionally exit after the first error with exit(1)
 	reuse_tmpc       bool // do not use random names for .tmp.c and .tmp.c.rsp files, and do not remove them
 	no_rsp           bool // when true, pass C backend options directly on the CLI (do not use `.rsp` files for them, some older C compilers do not support them)
-	no_std           bool // when true, do not pass -std=gnu99(linux)/-std=c99 to the C backend
+	no_std           bool // when true, do not pass -std=c99 to the C backend
 
 	no_parallel       bool // do not use threads when compiling; slower, but more portable and sometimes less buggy
 	parallel_cc       bool // whether to split the resulting .c file into many .c files + a common .h file, that are then compiled in parallel, then linked together.

